File: sast-python3/case/completeness/single_app_tracing/alias/alias_002_F.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = false
# evaluation item = 完整度->单应用跟踪完整度->别名
# scene introduction = 别名问题
# level = 2
# bind_url = completeness/single_app_tracing/alias/alias_002_F
# evaluation information end
import os


def alias_002_F(taint_src):
    a = {'value':taint_src }
    b = a
    b['value'] = '_'
    taint_sink(a['value'])


def taint_sink(o):
    os.system(o)


# 示例调用
if __name__ == "__main__":
    taint_src = "taint_src_value"
    alias_002_F(taint_src)

File: sast-js/case/completeness/single_app_tracing/alias/alias_002_F.js
// evaluation information start
// real case = false
// evaluation item = 完整度->单应用跟踪完整度->别名
// scene introduction = 别名问题
// level = 2
// bind_url = completeness/single_app_tracing/alias/alias_002_F
// evaluation information end
const { execSync } = require('child_process');


function alias_002_F(__taint_src) {
  let a = { value: "_" };
  let b = a;
  b.value = __taint_src;
  __taint_sink("safe");
}

function __taint_sink(o) {
  execSync(o);
}

const taint_src = "taint_src_value";

alias_002_F(taint_src);
